src/deadline/client/cli/_groups/manifest_group.py

```diff
@@ -228,28 +234,98 @@ def manifest_download(
 )
 @click.argument("manifest_file")
 @click.option("--profile", help="The AWS profile to use.")
-@click.option("--s3-cas-path", help="The path to the Content Addressable Storage root.")
+@click.option("--s3-cas-uri", help="The URI to the Content Addressable Storage S3 bucket and root.")
 @click.option(
-    "--farm-id", help="The AWS Deadline Cloud Farm to use. Alternative to using --s3-cas-path."
+    "--farm-id", help="The AWS Deadline Cloud Farm to use. Alternative to using --s3-cas-uri."
 )
 @click.option(
-    "--queue-id", help="The AWS Deadline Cloud Queue to use. Alternative to using --s3-cas-path."
+    "--queue-id", help="The AWS Deadline Cloud Queue to use. Alternative to using --s3-cas-uri."
 )
 @click.option("--json", default=None, is_flag=True, help="Output is printed as JSON for scripting.")
 @_handle_error
 def manifest_upload(
     manifest_file: str,
-    s3_cas_path: str,
+    s3_cas_uri: str,
     json: bool,
     **args,
 ):
-    raise NotImplementedError("This CLI is being implemented.")
+    # Input checking.
+    if not manifest_file or not os.path.isfile(manifest_file):
+        raise NonValidInputError(f"Specified manifest {manifest_file} does not exist. ")
+
+    # Where will we upload the manifest to?
+    required: set[str] = set()
+    if not s3_cas_uri:
+        required = {"farm_id", "queue_id"}
+
+    config = _apply_cli_options_to_config(required_options=required, **args)
+
+    # Logger
+    logger: ClickLogger = ClickLogger(is_json=json)
+
+    # Upload settings:
+    metadata: dict[str, Any] = {"Metadata": {}}
+    metadata["Metadata"]["file-system-location-name"] = manifest_file
+
+    bucket_name: str = ""
+    manifest_path: str = ""
+    session: boto3.Session = api.get_boto3_session(config=config)
+    if not s3_cas_uri:
+        farm_id = config_file.get_setting("defaults.farm_id", config=config)
+        queue_id = config_file.get_setting("defaults.queue_id", config=config)
+
+        deadline = api.get_boto3_client("deadline", config=config)
+        queue = deadline.get_queue(
+            farmId=farm_id,
+            queueId=queue_id,
+        )
+        bucket_name = queue["jobAttachmentSettings"]["s3BucketName"]
+        manifest_path = queue["jobAttachmentSettings"]["rootPrefix"]
+
+        # IF we supplied a farm and queue, use the queue credentials.
+        session = api.get_queue_user_boto3_session(
+            deadline=deadline,
+            config=config,
+            farm_id=farm_id,
+            queue_id=queue_id,
+            queue_display_name=queue["displayName"],
+        )
+
+    else:
+        # Self supplied cas path.
+        url_fragments = urlparse(s3_cas_uri)
+        bucket_name = url_fragments.netloc
+        manifest_path = url_fragments.path
```

Comment:
We could reuse existing function to parse the uri.

```suggestion
        s3_settings: JobAttachmentS3Settings = JobAttachmentS3Settings.from_s3_root_uri(s3_root_uri)
        bucket_name = s3_settings.s3BucketName
        root_prefix = s3_settings.rootPrefix
```

If this uri contains more than bucket name and root prefix, maybe the path logic can be added to the class.

Reply:
Let me try, I think that will work.

Reply:
Actually no that won't work, think of this case:

`s3://bucketname/prefix1/prefix2/some/path/i/want`

it is hard to capture `prefix1` and `prefix2` separate from `/some/path/i/want`